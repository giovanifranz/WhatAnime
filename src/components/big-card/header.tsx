'use client'

import Link from 'next/link'
import { ImArrowRight2 } from 'react-icons/im'

import { useWindowSize } from 'usehooks-ts'

import { Anime } from '@/store/schema'

import { Badge } from '../ui/badge'
import { Button } from '../ui/button'
import { CardHeader as UiCardHeader, CardTitle } from '../ui/card'

type Props = Pick<Anime, 'episodes' | 'title' | 'score' | 'id'>

export default function CardHeader({ episodes, title, score, id }: Props) {
  const { width } = useWindowSize()
  return (
    <UiCardHeader className="flex flex-row items-center justify-between p-0">
      <div className="flex h-14 max-w-[280px] flex-col justify-center">
        <CardTitle className="truncate text-lg ">{title}</CardTitle>
        <span>Episodes: {episodes}</span>
      </div>
      <div className="flex h-14 items-center gap-4">
        {width >= 768 && score ? (
          <div className="mt-4 flex h-12 flex-col items-center">
            <Badge>score</Badge>
            <span>{score}</span>
          </div>
        ) : null}
        <Button asChild className="md:w-32">
          <Link href={`/anime/${id}`}>
            {width < 768 ? <ImArrowRight2 size={20} /> : 'Go to Page'}
          </Link>
        </Button>
      </div>
    </UiCardHeader>
  )
}
