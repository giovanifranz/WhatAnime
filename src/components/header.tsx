import { Logo } from "./logo";

export function Header() {
  return (
    <header className="p-4 bg-neutral-800 fixed top-0 left-0 right-0 z-50">
      <div className="mx-auto max-w-7xl">
        <Logo />
      </div>
    </header>
  );
}
