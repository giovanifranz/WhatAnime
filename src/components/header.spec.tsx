import { render, screen } from '@/test/test-utils'

import { Header } from './header'

describe('Header', () => {
  it('Deve renderizar o Header corretamente', () => {
    const { asFragment } = render(<Header />)

    expect(asFragment()).toMatchSnapshot()
    expect(screen.getByTestId('logo')).toBeVisible()
  })
})
