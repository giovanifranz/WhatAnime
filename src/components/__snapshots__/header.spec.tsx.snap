// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > Deve renderizar o Header corretamente 1`] = `
<DocumentFragment>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('dark')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'dark'}catch(e){}}()
  </script>
  <header
    class="absolute inset-x-0 top-0 z-50 bg-neutral-800 p-4"
  >
    <div
      class="mx-auto max-w-7xl"
    >
      <a
        href="/"
      >
        <img
          alt="WhatAnime"
          data-nimg="1"
          decoding="async"
          height="76"
          loading="lazy"
          src="/src/assets/logo.svg"
          style="color: transparent;"
          width="227"
        />
      </a>
    </div>
  </header>
</DocumentFragment>
`;
